bundles/statistics/statsgrid2016/components/IndicatorParametersHandler.js

```diff
@@ -0,0 +1,133 @@
+    this.locale = locale;
+    this._values = null;
+    this.datasource = null;
+    this.indicators = null;
+    this.elements = null;
+
+    Oskari.makeObservable(this);
+}, {
+    getData: function ( datasrc, indicators, elements ) {
+
+        this.datasource = datasrc;
+        this.indicators = indicators;
+        this.elements = elements;
+
+        if ( Array.isArray( indicators ) ) {
+            this.handleMultipleIndicators(datasrc, indicators, elements);
+            return;
+        }
+        this.handleSingleIndicator(datasrc, indicators, elements);
+    },
+    handleSingleIndicator: function( indId, cb) {
+        var me = this;
+        var errorService = me.service.getErrorService();
+        var locale = this.locale;
+        var panelLoc = locale.panels.newSearch;
+
+        this.service.getIndicatorMetadata(this.datasource, indId, function (err, indicator) {
+            if (me.elements.dataLabelWithTooltips) {
+                me.elements.dataLabelWithTooltips.find('.tooltip').hide();
+            }
+            if (err) {
+                // notify error!!
+                errorService.show(locale.errors.title, locale.errors.indicatorMetadataError);
+                return;
+            }
+
+            // selections
+            var selections = [];
+            var combinedValues = {};
+
+            indicator.selectors.forEach(function (selector, index) {
+                selector.allowedValues.forEach(function (val) {
+                    if ( !combinedValues[selector.id] ) {
+                        combinedValues[selector.id] = [];
+                    }
+
+                    var name = val.name || val.id || val;
+                    val.title = val.name;
+                    var optName = (panelLoc.selectionValues[selector.id] && panelLoc.selectionValues[selector.id][name]) ? panelLoc.selectionValues[selector.id][name] : name;
+
+                    var valObject = {
+                        id: val.id || val,
+                        title: optName
+                    };
+                    combinedValues[selector.id].push(valObject);
+                    selections.push(valObject);
+                });
+            });
+
+            if (indicator.regionsets.length === 0) {
+                errorService.show(locale.errors.title, locale.errors.regionsetsIsEmpty);
+            }
+
+            // Add margin if there is selections
+            if (indicator.selectors.length > 0) {
+                //regionSelect.container.addClass('margintop');
+            } else {
+                errorService.show(locale.errors.title, locale.errors.indicatorMetadataIsEmpty);
+            }
+
+            var data = {
+                datasrc: me.datasource,
+                indicators: me.indicators,
+                selectors: selections,
+                regionset: indicator.regionsets,
+                values: combinedValues
+            }
+            cb(data);
+            //me.trigger('Data.Loaded', data);
```

Comment:
With single indicator the trigger isn't sent at all -> the logic is broken. Either do a:

    if (typeof cb === 'function') {
        cb(data);
    } else {
        me.trigger('Data.Loaded', data);
    }

Or move the trigger('Data.Loaded') to the constructor calling this function.